### PR TITLE
Fix: [UI] Error not showing in Task Group view Bug1774

### LINKED ISSUE #1774: [UI] Error not showing in Task Group view
**To Reproduce**

1. Go to https://taskcluster-ui.herokuapp.com/tasks/groups/ZeW7IKcDRQipFT2TN4UAcg

**Actual behavior**

An empty view

Screen Shot 2019-10-28 at 11 42 32 AM

**Expected behavior**
An error ``` No task-group with taskGroupId: `ZeW7IKcDRQipFT2TN4UAcg` ``` shows up. The error message to display will be in the response of the network request.

Screen Shot 2019-10-28 at 11 43 45 AM

### PR DESCRIPTION
**Pull request by somchi**
fix: solving issue with [UI] error not showing in task group view in  https://taskcluster-ui.herokuapp.com/tasks/groups/ZeW7IKcDRQipFT2TN4UAcg

Fixes #1774.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 node_modules
 .eslintcache
+.vscode
```
Comment (vishakhanihore) [inline .gitignore]: Remove this too.

File: changelog/bug-1750.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+level: silent
+reference: bug 1750
+---
```
Comment (vishakhanihore) [inline changelog/bug-1750.md]: No need for two changelogs.

File: ui/src/components/ProvisionerDetailsCard/index.jsx [hunk 3]
```diff
@@ -30,8 +30,8 @@ import { ACTION_CONTEXT } from '../../utils/constants';
     fontSize: '0.7rem',
   },
   headline: {
-    paddingLeft: theme.spacing.triple,
-    paddingRight: theme.spacing.triple,
+    paddingLeft: theme.spacing.double,
```
Comment (vishakhanihore) [inline ui/src/components/ProvisionerDetailsCard/index.jsx]: Please see the `Migrating to MUI 4.0 PR` this terminology has been depreciated and would result in conflicts after the PR is merged

File: .gitignore (hunk 1 above)
Comment (somchi) [reply .gitignore]: alright

File: changelog/bug-1750.md (hunk 2 above)
Comment (somchi) [reply changelog/bug-1750.md]: will fix that right away

File: ui/src/components/ProvisionerDetailsCard/index.jsx (hunk 3 above)
Comment (somchi) [reply ui/src/components/ProvisionerDetailsCard/index.jsx]: where do i find that to read up

Comment (vishakhanihore) [reply ui/src/components/ProvisionerDetailsCard/index.jsx]: You can see #1729 for the replacement

Comment (somchi) [reply ui/src/components/ProvisionerDetailsCard/index.jsx]: but that update has not been added to the master branch. it is still on the previous version of the material ui

Comment (helfi92) [reply ui/src/components/ProvisionerDetailsCard/index.jsx]: Correct. We can't use material-ui v4 api at the moment. We will need to rely on `theme.spacing.double` for now.

File: ui/src/components/ProvisionerDetailsCard/index.jsx [hunk 4]
```diff
@@ -30,8 +30,8 @@ import { ACTION_CONTEXT } from '../../utils/constants';
...
+    paddingRight: theme.spacing.double,
```
Comment (helfi92) [inline ui/src/components/ProvisionerDetailsCard/index.jsx]: This change is out of scope for this PR and I think it's already been handled by another PR. We can remove these style changes.

File: ui/src/views/Tasks/TaskGroup/index.jsx [hunk 5]
```diff
@@ -591,12 +590,12 @@ export default class TaskGroup extends Component {
...
-      error.graphQLErrors[0] &&
-      error.graphQLErrors[0].statusCode === 404 &&
-      error.graphQLErrors[0].requestInfo.method === 'task'
-    );
+
+    // return (
+    //   error.graphQLErrors &&
+    //   error.graphQLErrors[0].statusCode === 404 &&
+    //   error.graphQLErrors[0].requestInfo.method === 'task'
+    // );
```
Comment (helfi92) [inline ui/src/views/Tasks/TaskGroup/index.jsx]: Instead of commenting this code out, we should iterate over the array and check if there are any entry errors that fit the conditions (i.e., `statusCode` is `404` and method is 'task'). We can probably use the [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) to do this.

Comment (somchi) [reply ui/src/views/Tasks/TaskGroup/index.jsx]: alright will work on that

File: ui/src/views/Tasks/TaskGroup/index.jsx [hunk 6]
```diff
@@ -588,15 +587,12 @@ export default class TaskGroup extends Component {
...
     // Task groups do not necessarily have a decision task,
     // so handle task-not-found errors gracefully
-    return (
-      error.graphQLErrors &&
-      error.graphQLErrors[0] &&
-      error.graphQLErrors[0].statusCode === 404 &&
-      error.graphQLErrors[0].requestInfo.method === 'task'
-    );
+    const errorStatus = error.graphQLErrors.some((error, index) => {
+      return index == 0 && error.statusCode === 404 && error.requestInfo.method === 'task'
```
Comment (helfi92) [inline ui/src/views/Tasks/TaskGroup/index.jsx]: Is there a reason why we are only checking for index 0?

Comment (somchi) [reply ui/src/views/Tasks/TaskGroup/index.jsx]: and really yet to know the reason for that

Comment (helfi92) [reply ui/src/views/Tasks/TaskGroup/index.jsx]: I'm not sure I understand. Do you mind elaborating, please? :)

Comment (somchi) [reply ui/src/views/Tasks/TaskGroup/index.jsx]: so i noticed the ErrorPanel displays only the error at index 0, reasons that was done i don't really know yet. That is why i am factoring that in too

Comment (helfi92) [reply ui/src/views/Tasks/TaskGroup/index.jsx]: The error panel displays the first error it gets if it receives an object, but if it's a string, it will display it as is.

The errors we get from graphql are not always in the same order. For example, it's possible that index 1 has status code 404 and requestInfo "task". We'll need to make sure we catch it here. The current logic only seems to look at index 0 which defeats the purpose of using `.some`.

**Comment by somchi** [comment]
i ran yarn lint --fix, and it was clear, will work on that

**Comment by somchi** [comment]
Hi, Hassan i haven't heard from you over this issues. i have been dying to hear from you

**Comment by djmitche** [comment]
@somchi remember that the Taskcluster staff are all working on this project as a profession, so we take weekends off :)

Hassan (like all of us) is very busy and I'm sure will get to this soon.  However, CI has already given you some feedback -- I see a number of red "X"'s in the checks.  So, addressing those would be a good place to start.

**Comment by helfi92** [comment]
To fix the linting errors, you can run `yarn lint --fix` inside the **`ui/`** directory.

**Comment by somchi** [comment]
Have i been able to resolve this issue @helfi92
